internal/resolver/dns/dns_resolver.go

```diff
@@ -201,18 +207,49 @@ func (d *dnsResolver) Close() {
 
 func (d *dnsResolver) watcher() {
 	defer d.wg.Done()
+	backoffIndex := 1
+	nextBackoffRate := backoff.DefaultExponential.Backoff(backoffIndex)
 	for {
 		select {
 		case <-d.ctx.Done():
+			d.pollTimer.Stop()
 			return
 		case <-d.rn:
+			// Reset polling timer if active, as ResolveNow was called from ClientConn.
+			if !d.pollTimer.Stop() {
+				select {
+				case <-d.pollTimer.C:
+				default:
+				}
+			}
+			backoffIndex = 1
```

Comment:
If it fails below, I think we want to keep our spot in the exponential backoff, not reset.

Reply:
I feel like a resolveNow call should reset it? Logically, if exponentially backing off is tied to a resolvenow call failing (equivalent to the initial newtimer with duration 0), why would we persist exponetial backoff index? That doesn't make sense to me.

Reply:
Discussed in chat, changed to move to next iteration.